data/modules/TradeShips/Trader.lua

```diff
@@ -0,0 +1,355 @@
+-- Copyright © 2008-2021 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local e = require 'Equipment'
+local Engine = require 'Engine'
+local Game = require 'Game'
+local Ship = require 'Ship'
+local ShipDef = require 'ShipDef'
+local Space = require 'Space'
+local Timer = require 'Timer'
+
+local Core = require 'modules.TradeShips.Core'
+
+local Trader = {}
+-- this module contains functions that work for single traders
+
+Trader.addEquip = function (ship)
+	local ship_type = ShipDef[ship.shipId]
+
+	-- add standard equipment
+	ship:AddEquip(e.hyperspace['hyperdrive_'..tostring(ship_type.hyperdriveClass)])
+	if ShipDef[ship.shipId].equipSlotCapacity.atmo_shield > 0 then
+		ship:AddEquip(e.misc.atmospheric_shielding)
+	end
+	ship:AddEquip(e.misc.radar)
+	ship:AddEquip(e.misc.autopilot)
+	ship:AddEquip(e.misc.cargo_life_support)
```

Comment:
I don't see any provisions here made for cargo capacity. Is the intent that when a ship is full, AddEquip will just silently fail? Or is this made on the assumption that this function will always be operating on ships with enough capacity to hold all of these modules and cargo besides?

Reply:
@Web-eWorks it is assumed that this function will always run on an empty ship, It seems to work well for all ships. I think that the equipment of the ship is quite a broad topic, so I most likely did not even touch this function. It seems to me that for each ship model an optimal set of equipment should be designed for different purposes (roles).

Reply:
added an assertion that the cargo hold is empty

Reply:
The idea of a default loadout makes sense and is something I've been wanting to look into for a bit; though the logistics of passing the data around is something that I think needs a bit more thought than trying to tack it on to ShipDef.

I was mostly asking if there were conditions where a small ship might get chosen and potentially only have room for a hyperdrive and a few (<10) tons of cargo; though if there's code above this function that's only selecting ships that are intended to be larger cargo ships then this isn't an issue at all.